pkg/services/auth/github_test.go

```diff
@@ -91,57 +111,86 @@ var _ = Describe("Github Device Flow", func() {
 		Expect(cliOutput.String()).To(ContainSubstring(verificationUri))
 	})
 
-	// These auth flow tests are failing intermittently, so bypass them for now
-	XDescribe("pollAuthStatus", func() {
-		It("retries after a slow_down response from github", func() {
-			rt := newMockRoundTripper(3, token)
-			client.Transport = &testServerTransport{testServeUrl: ts.URL, roundTripper: rt}
-			interval := 5 * time.Second
+	Describe("pollAuthStatus", func() {
+		var rt *mockAuthRoundTripper
+		var s *sleeper
+
+		// pollTimes is a convenience function to convert from a series of polling intervals
+		// to their respective polling timestamps, relative to the sleeper type's starting time
+		pollTimes := func(intervals []time.Duration) []time.Time {
+			zero := time.Time{}
+			times := make([]time.Time, len(intervals))
+			for index, interval := range intervals {
+				switch index {
+				case 0:
+					times[index] = zero.Add(interval)
+				default:
+					times[index] = times[index-1].Add(interval)
+				}
+			}
+			return times
+		}
 
-			c := clock.NewMock()
+		drainPollTimes := func(pollChan <-chan time.Time) (result []time.Time) {
```

Comment:
We will probably need to decide as a team if we like this "named return variable" pattern. I have always found it an odd choice for the language and I have a feeling there will be mixed opinions on whether or not we use it.

Until then, so long as it passes the linter, :+1:

Reply:
Fair point. I wouldn’t use named return variables in a longer function; it seemed worth the trade-off here, but I’ll be happy to go with the team consensus if we decide to ditch the pattern entirely.